src/components/Home/Home.jsx

```diff
@@ -1,27 +1,73 @@
 import React from 'react';
-import { Redirect, useHistory } from 'react-router-dom';
+import { useState } from 'react';
+import { useForm } from '../../hooks/useForm';
+import { useFirebase } from '../../hooks/useFirebase.js';
+import { useHistory } from 'react-router-dom';
 import getToken from '../../lib/tokens';
 
 const Home = () => {
-  const userToken = localStorage.getItem('token');
+  const [values, handleInputChange, setValues] = useForm({
+    token: '',
+  });
+
+  const [msn, setMsn] = useState('');
+  const [error, setError] = useState('');
+  const [load, setLoad] = useState('');
+  const { getAll } = useFirebase();
+
+  function searching() {
+    setLoad('searching...');
+
+    var docRef = getAll().doc(values.token);
+
+    docRef
+      .get()
+      .then((doc) => {
+        if (doc.exists) {
+          setMsn('The list is found');
+          setLoad('');
+          localStorage.setItem('token', values.token);
+          history.push('/list');
+        } else {
+          setMsn('The list is not found');
+          setLoad('');
+        }
+      })
+      .catch((error) => {
+        setError(`Error getting document: ${error}`);
+      });
+  }
 
   const history = useHistory();
```

Comment:
what do you think if put this line at the same level where you declared the others hooks :D

Reply:
Thanks, it has been solve.👍🏻

Reply:
Just wanted to add to this comment a little bit. I really like the idea of putting hooks next to each other all at the top of the component because it helps to ensure that we're following the "rules-of-hooks". Specifically, hooks should never be called conditionally. Having them towards the top of the component helps to ensure that you don't accidentally end up with a situation in which a hook isn't called because you returned early from the component. 
Great comment Fauricio!

https://reactjs.org/docs/hooks-rules.html